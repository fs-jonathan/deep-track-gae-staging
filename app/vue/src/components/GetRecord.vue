<template>
  <div class="getRecord">
    <div id="loader">
      <vue-element-loading :active="loading" spinner="bar-fade-scale" :is-full-screen="true"/>
    </div>

    <div class="bg-blue-lightest border-t border-b border-blue text-blue-dark px-4 py-3" v-if="error">
      <p class="text-sm">{{ error }}</p>
    </div>

    <div id="calendar" class='control border-b m-3'>
      <v-date-picker
        mode='range'
        v-model='selectedDate'
        :input-props='{ class: "input flex", readonly: true, style: "min-width: 300px;" }'
        show-caps>
        <b-field>
          <b-input
            type='text'
            icon='calendar'
            :value='inputValue'
            rounded>
          </b-input>
        </b-field>
      </v-date-picker>
    </div>

    <div style="display:none" id="content" v-if="results">
      <div v-for="(result, key, index) in results" :key="index">
        <row :result="result" />
      </div>
    </div>
  </div>
</template>

<script>
import axios from 'axios'
import moment from 'moment'
import BaseRow from './BaseRow'

export default {
  name: 'GetRecord',
  data: () => ({
    loading: false,
    results: null,
    error: null,
    selectedDate: {
      start: moment(Date.now()).subtract(5, 'd').toDate(),
      end: new Date()
    }
  }),
  components: {
    'row': BaseRow,
  },
  created () {
    // fetch the data when the view is created and the data is already being observed
    this.getJson()
  },
  methods: {
    getJson: function() {
      this.loading = true

      axios.post('/getReport')
           .then(response => {
             this.results = response.data;
           })
           .catch(err => {
             this.error = err.toString();
           })
           .finally(() => {
             this.loading = false
             document.getElementById("content").style.display = "block";
           })
    }
  }
}
</script>
