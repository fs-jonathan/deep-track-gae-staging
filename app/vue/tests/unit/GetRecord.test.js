import { shallowMount, createLocalVue } from '@vue/test-utils'
import GetRecord from '@/components/GetRecord.vue'
import VueElementLoading from 'vue-element-loading'
import VCalendar from 'v-calendar'
import BootstrapVue from 'bootstrap-vue'
import flushPromises from 'flush-promises'

const localVue = createLocalVue()
localVue.use(VCalendar)
localVue.use(BootstrapVue)
localVue.component('VueElementLoading', VueElementLoading)

const data = [{"id":1,"title":"本日","subtitle":"（現時点まで）","cost":82,"compare":0,"rate":0},{"id":2,"title":"昨日","subtitle":"先週と同じ曜日との比較","cost":14,"compare":0.1,"rate":0.1},{"id":3,"title":"今月（現時点まで）","subtitle":"先月の同じ日との比較","cost":1517,"compare":0,"rate":0},{"id":4,"title":"今月（現時点まで）","subtitle":"先月の同じ日との比較","cost":0,"compare":0,"rate":0},{"id":5,"title":"全期間","subtitle":"","cost":1517,"compare":0,"rate":0}]

describe('GetRecord.vue', () => {
  it('renders correctly', async () => {
    const msg = 'new message'
    const wrapper = shallowMount(GetRecord, {
      attachToDocument: true,
      propsData: { result: {
        title: msg,
        cost: 0,
        id: 2
      } },
      localVue,
      stubs: {
        BField: true
      },
      mocks: {
        $axios: {
          post: () => Promise.resolve({ data: data })
        }
      }
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
    await flushPromises()
    expect(wrapper.element).toMatchSnapshot()
  })
})
