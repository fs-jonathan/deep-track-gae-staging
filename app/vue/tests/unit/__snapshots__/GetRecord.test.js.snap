// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetRecord.vue renders correctly 1`] = `
<div
  class="getRecord"
  result="[object Object]"
>
  <div
    id="loader"
  >
    <vueelementloading-stub
      isfullscreen="true"
      spinner="bar-fade-scale"
    />
  </div>
   
  <!---->
   
  <div
    class="control border-b m-3"
    id="calendar"
  >
    <vdatepicker-stub
      inputprops="[object Object]"
      mode="range"
      popoveralign="left"
      popovercontentoffset="10"
      popoverdirection="bottom"
      popovervisibility="hover"
      showcaps="true"
      showdaypopover="true"
      themestyles="[object Object]"
      tintcolor="#66B3CC"
      value="[object Object]"
    >
      <bfield-stub>
        <binput-stub
          icon="calendar"
          rounded=""
          type="text"
        />
      </bfield-stub>
    </vdatepicker-stub>
  </div>
   
  <div
    id="content"
    style="display: block;"
  >
    <div>
      <row-stub
        result="[object Object]"
      />
    </div>
    <div>
      <row-stub
        result="[object Object]"
      />
    </div>
    <div>
      <row-stub
        result="[object Object]"
      />
    </div>
    <div>
      <row-stub
        result="[object Object]"
      />
    </div>
    <div>
      <row-stub
        result="[object Object]"
      />
    </div>
  </div>
</div>
`;
